### PR TITLE
Clean up extensions

### LINKED ISSUE #1157: inconsistent file names for extensions
image

Should standardize on one pattern and stick to it

### PR DESCRIPTION
**Pull request by Kharchevskyi**
This PR clean up extensions in app. 
Move all extension to FlowCryptCommon
Move BigInt, Toast, MBProgressHud to FlowCryptCommon

close #1157
Screenshot 2021-12-28 at 15 03 07

----------------------------------

**Tests** _(delete all except exactly one)_:
- Does not need tests (refactor only, docs or internal changes)
--------------------------------

### To be filled by reviewers

I have reviewed that this PR... _(tick whichever items you personally focused on during this review)_:
- [ ] addresses the issue it closes (if any)
- [ ] code is readable and understandable
- [ ] is accompanied with tests, or tests are not needed
- [ ] is free of vulnerabilities
- [ ] is documented clearly and usefully, or doesn't need documentation

### CODE REVIEW

File: FlowCryptCommon/Extensions/CalendarExtensions.swift [hunk 1]
```diff
@@ -9,8 +9,15 @@
...
-        return self.isDate(date, equalTo: Date(), toGranularity: .year)
+        self.isDate(date, equalTo: Date(), toGranularity: .year)
+    }
+}
+
+public extension Bundle {
+    
+    static var flowCryptBundleTypeExtension: String {
+        guard let bundleIdentifier = Bundle.main.bundleIdentifier else { return "aaa" }
+        return bundleIdentifier ?? "N?A"
```
Comment (sosnovsky) [inline FlowCryptCommon/Extensions/CalendarExtensions.swift:21]: It seems this property isn't used anywhere, so can be deleted, I think
(also `aaa` is a bit strange return value :)

**Review by sosnovsky — CHANGES_REQUESTED** [review]
There are also a couple conflicts with `master` branch

File: FlowCryptCommon/Extensions/CalendarExtensions.swift (hunk 1 above)
Comment (Kharchevskyi) [reply FlowCryptCommon/Extensions/CalendarExtensions.swift]: Merged master and replaced test code. Check latest commit

File: FlowCryptCommon/Extensions/StringExtensions.swift [hunk 2]
```diff
@@ -61,6 +65,36 @@ public extension NSAttributedString {
...
+    }
+    
+    var recipientDomain: String? {
+        let parts = self.split(separator: "@")
+        if parts.count != 2 {
+            return nil
+        }
+        return String(parts[1])
+    }
+
```
Comment (sosnovsky) [inline FlowCryptCommon/Extensions/StringExtensions.swift:97]: `userAndRecipientDomain`, `userEmail` and `recipientDomain` properties were replaced by the single `emailParts` property, so they can be deleted

**Review by sosnovsky — CHANGES_REQUESTED** [review]

**Review by sosnovsky — APPROVED** [review]

**Pull request merged**